src/main/java/net/guides/springboot/todomanagement/service/StudentService.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * 
+ */
+package net.guides.springboot.todomanagement.service;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import javax.validation.Valid;
+
+import org.apache.commons.io.FileUtils;
+import org.json.simple.parser.ParseException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.stereotype.Service;
+import org.thymeleaf.TemplateEngine;
+import org.thymeleaf.context.Context;
+
+import com.google.gson.Gson;
+
+import lombok.extern.slf4j.Slf4j;
+import net.guides.springboot.todomanagement.model.Student;
+import net.guides.springboot.todomanagement.repository.StudentRepository;
+
+/**
+ * @author elect
+ *
+ */
+@Service
+@EnableAsync
+@Slf4j
+public class StudentService {
+	/**
+	 * Inject to StudentRepository
+	 */
+	@Autowired
+	private StudentRepository studentRepository;
+	/**
+	 * file directory path
+	 */
+	@Value("${file.upload-dir}")
+	private String path;
+	/**
+	 * templateEngine
+	 */
+	private final TemplateEngine templateEngine;
+	/**
+	 * javaMailSender
+	 */
+	private final JavaMailSender javaMailSender;
+
+	/**
+	 * creating the student
+	 * 
+	 * @param student
+	 * @return
+	 */
+	public Student createStudent(final Student student) {
+		log.info("Start of StudentService :: createStudent ");
+		Student saveStudent = null;
+		if (student.getStudentEmail() != null) {
+			saveStudent = studentRepository.save(student);
+		}
```

Comment:
This is a very dangerous code. This method will always be successful with out throwing any exception even if the condition is satisfied or not.

Reply:
fixed